components/hab/src/main.rs

```diff
@@ -1261,22 +1258,8 @@ async fn sub_svc_load(svc_load: SvcLoad) -> Result<()> {
 }
 
 async fn sub_svc_bulk_load(svc_bulk_load: SvcBulkLoad) -> Result<()> {
-    let default_svc_load = ConfigOptSvcLoad::from_default_config_files()?;
-    for path in svc_bulk_load.svc_config_paths {
-        for entry in WalkDir::new(path) {
-            let entry = entry?;
-            let path = entry.path();
-            if entry.file_type().is_file() {
-                if let Some(extension) = path.extension() {
-                    if extension == "toml" {
-                        let mut svc_load = configopt::from_toml_file(path)?;
-                        // Patch the svc load with values from the default svc load
-                        default_svc_load.clone().patch_for(&mut svc_load);
-                        sub_svc_load(svc_load).await?;
-                    }
-                }
-            }
-        }
+    for svc_load in svc::svc_loads_from_paths(&svc_bulk_load.svc_config_paths)? {
+        sub_svc_load(svc_load).await?;
```

Comment:
If one of these loads fails for whatever reason, you could end up with some number of services that also didn't load, only because the first one didn't. Since this is really just the "send the Supervisor the command" operation, and not a synchronous "wait for the service to actually be running" operation, that's _probably_ OK, but it might be nice to indicate to the user which services' load commands got through, which one failed, and which ones didn't even get sent.

Reply:
Addressed in [87c005](https://github.com/habitat-sh/habitat/pull/7765/commits/87c005215f3c0ac3a1d18a57a31fad3ded81b1c9). It now tries to load all services and then presents an error message for each failed load. Thanks for pointing this out!